pkg/query-service/app/logs/v4/query_builder.go

```diff
@@ -29,3 +35,492 @@ const (
+	bucketEnd := logsEnd / NANOSECOND
+
+	// timestamp filter , bucket_start filter is added for primary key
+	timeFilter := fmt.Sprintf("(timestamp >= %d AND timestamp <= %d) AND (ts_bucket_start >= %d AND ts_bucket_start <= %d)", logsStart, logsEnd, bucketStart, bucketEnd)
+
+	// build the where clause for main table
+	filterSubQuery, err := buildLogsTimeSeriesFilterQuery(mq.Filters, mq.GroupBy, mq.AggregateAttribute)
+	if err != nil {
+		return "", err
+	}
+	if filterSubQuery != "" {
+		filterSubQuery = " AND " + filterSubQuery
+	}
+
+	// build the where clause for resource table
+	resourceSubQuery, err := buildResourceSubQuery(bucketStart, bucketEnd, mq.Filters, mq.GroupBy, mq.AggregateAttribute)
+	if err != nil {
+		return "", err
+	}
+	// join both the filter clauses
+	if resourceSubQuery != "" {
+		filterSubQuery = filterSubQuery + " AND (resource_fingerprint GLOBAL IN " + resourceSubQuery + ")"
+	}
+
+	// get the select labels
+	selectLabels := getSelectLabels(mq.AggregateOperator, mq.GroupBy)
+
+	// get the order by clause
+	orderBy := orderByAttributeKeyTags(panelType, mq.OrderBy, mq.GroupBy)
+	if panelType != v3.PanelTypeList && orderBy != "" {
+		orderBy = " order by " + orderBy
+	}
+
+	// if noop create the query and return
+	if mq.AggregateOperator == v3.AggregateOperatorNoOp {
+		// with noop any filter or different order by other than ts will use new table
+		sqlSelect := constants.LogsSQLSelectV2
+		queryTmpl := sqlSelect + "from signoz_logs.%s where %s%s order by %s"
+		query := fmt.Sprintf(queryTmpl, DISTRIBUTED_LOGS_V2, timeFilter, filterSubQuery, orderBy)
+		return query, nil
+		// ---- NOOP ends here ----
+	}
+
+	// ---- FOR aggregation queries ----
+
+	// get the having conditions
+	having := logsV3.Having(mq.Having)
+	if having != "" {
+		having = " having " + having
+	}
+
+	// get the group by clause
+	groupBy := logsV3.GroupByAttributeKeyTags(panelType, graphLimitQtype, mq.GroupBy...)
+	if panelType != v3.PanelTypeList && groupBy != "" {
+		groupBy = " group by " + groupBy
+	}
+
+	// get the aggregation key
+	aggregationKey := ""
+	if mq.AggregateAttribute.Key != "" {
+		aggregationKey = getClickhouseKey(mq.AggregateAttribute)
+	}
+
+	// for limit queries, there are two queries formed
+	// in the second query we need to add the placeholder so that first query can be placed
+	if graphLimitQtype == constants.SecondQueryGraphLimit {
+		filterSubQuery = filterSubQuery + " AND " + fmt.Sprintf("(%s) GLOBAL IN (", logsV3.GetSelectKeys(mq.AggregateOperator, mq.GroupBy)) + "#LIMIT_PLACEHOLDER)"
+	}
+
+	aggClause, err := generateAggregateClause(mq.AggregateOperator, aggregationKey, step, preferRPM, timeFilter, filterSubQuery, groupBy, having, orderBy)
+	if err != nil {
+		return "", err
+	}
+
+	var queryTmplPrefix string
+	if graphLimitQtype == constants.FirstQueryGraphLimit {
+		queryTmplPrefix = "SELECT"
+	} else if panelType == v3.PanelTypeTable {
+		queryTmplPrefix =
+			"SELECT now() as ts,"
```

Comment:
This `SELECT now() as ts` is no longer needed. The get time series result function on reader works without ts column.

Reply:
Sure.